custom_connectors/custom_auth/kizen.rb

```diff
@@ -632,8 +858,312 @@
+        [
+          { name: 'id'}
+        ]
+      end,
+    },
+
+  add_custom_lead_source: { # This is new here
+    title: 'Add Custom Lead Source to a Contact',
+    subtitle: 'Add Custom Lead Source to a Contact',
+    description: lambda do
+      "Add <span class='provider'>Custom Lead Source to a Contact"
+    end,
+
+    input_fields: lambda do
+      [
+        { name: 'client_id', Label: 'Contact ID', optional: false },
+        { name: 'source', 
+          Label: 'Custom Lead Source', 
+          control_type: 'select',
+          pick_list: 'lead_sources',
+          hint: 'To add Custom Lead Sources,
+          navigate inside a contact record and add a custom source',
+          optional: false },
+        { name: 'campaign', Label: 'Campaign Name', optional: true },
+        { name: 'medium', Label: 'Medium', optional: true },
+        { name: 'term', Label: 'Term', optional: true },
+        { name: 'content', Label: 'Content', optional: true }
+      ]
+    end,
+    execute: lambda do |_connection, input|
+      post('https://app.kizen.com/api/lead-source-custom-source')
+           .payload(
+             'client': input['client_id'],
+             'source': input['source'],
+             'campaign': input['campaign'],
+             'medium': input['medium'],
+             'term': input['term'],
+             'content': input['content']
+           )
+    end,
+
+    output_fields: lambda do
+      [
+        { name: 'id' },
+        { name: 'client', Label: 'Contact ID' }
+      ]
+    end
   },
+log_activity: { # This is new
+  title: 'Log an Activity',
+  subtitle: 'Log an Activity',
+  description: lambda do
+    "Log <span class='provider'>an activity in Kizen"
+  end,
+
+  input_fields: lambda do |object_definitions|
+    object_definitions['log_activity_input']
+  end,
+  execute: lambda do |_connection, input|
+    format_payload = call('format_payload', input)
+    payload = format_payload.map do |key, value|
+      if key.include?('custom_fields')
+        custom_fields = value&.map do |k, v|
+          { k.split('~').last =>
+              {
+                'field_id' => k.split('~').first,
+                'value' => v
+              } }
+        end&.inject(:merge)
+        { 'custom_fields' => custom_fields }
+      else
+        { key => value }
+      end
+    end.inject(:merge)
+    result = post('/api/logged-activity', payload)
+    formatted_response =
+      result.map do |key, value|
+        if key.include?('custom_fields')
+          custom_fields = value.values&.map do |object|
+            { object['field_id'] => object['value'] }
+          end&.inject(:merge)
+          { 'custom_fields' => custom_fields }
+        else
+          { key => value }
+        end
+      end&.inject(:merge)
+    call('format_response', formatted_response.compact)
+  end,
+
+  output_fields: lambda do
+    [
+      { name: 'id' }
+    ]
+  end
+},
+find_contact_by_email: { # This is new
```

Comment:
Can we group `Find contact by email`,`Find contact by id`, `Find company by name` and `Find deal by name` into a single `Search object` action? This can be easily done by following the patterns in the `create_object` and `update_object` steps

It could be `Search objects` with object selection for
1. Contact by ID
2. Contact by email
3. Company by name
4. Deal by name

Reply:
I don't really want to do this. It is important to have all possible actions shown to the end user on the very first screen(https://www.workato.com/integrations/kizen) so they can see how many actions and triggers are possible.